yt_dlp/extractor/cbc.py

```diff
@@ -228,6 +229,63 @@ def _real_extract(self, url):
         }
 
 
+class CBCPlayerPlaylistIE(InfoExtractor):
+    IE_NAME = 'cbc.ca:player:playlist'
+    _VALID_URL = r'(?:cbcplayer:|https?://(?:www\.)?cbc\.ca/(?:player/)(?!play/))(?P<id>.+)'
+    _TESTS = [{
+        'url': 'https://www.cbc.ca/player/news/TV%20Shows/The%20National/Latest%20Broadcast',
+        'playlist_mincount': 25,
+        'info_dict': {
+            'id': 'news/tv shows/the national/latest broadcast',
+            'ie_key': 'CBCPlayer',
+        }
+    }, {
+        'url': 'https://www.cbc.ca/player/news/Canada/North',
+        'playlist_mincount': 25,
+        'info_dict': {
+            'id': 'news/canada/north',
+            'ie_key': 'CBCPlayer',
+        }
+    }]
+
+    def _real_extract(self, url):
+        # We have no other playlist id other than the URL, so use that
+        playlist_id = self._match_id(url).replace('%20', ' ').lower()
+        # the json info we're looking for isn't marked as such, therefore we need a bit of a workaround.
+        json_content = self._parse_json(
+            re.sub(
+                r'^.*?{', '{',
+                get_element_by_id('initialStateDom', self._download_webpage(url, playlist_id))
+            )[:-1],
+            playlist_id)
```

Comment:
use `_search_json()`
```suggestion
        webpage = self._download_webpage(url, playlist_id)
        json_content = self._search_json(
            r'window\.__INITIAL_STATE__\s*=', webpage, 'initial state', playlist_id)
```